service/worker/migration/activities.go

```diff
@@ -482,3 +540,194 @@ func (a *activities) SeedReplicationQueueWithUserDataEntries(ctx context.Context
 		activity.RecordHeartbeat(ctx, heartbeatDetails)
 	}
 }
+
+func (a *activities) createReplicationTasks(ctx context.Context, request *genearteAndVerifyReplicationTasksRequest, detail *replicationTasksHeartbeatDetails) error {
+	rateLimiter := quotas.NewRateLimiter(request.RPS, int(math.Ceil(request.RPS)))
+
+	for i := 0; i < len(request.Executions); i++ {
+		r := &detail.Results[i]
+		if r.isCompleted() {
+			continue
+		}
+
+		we := request.Executions[i]
+		tags := []tag.Tag{tag.WorkflowType(forceReplicationWorkflowName), tag.WorkflowNamespaceID(request.NamespaceID), tag.WorkflowID(we.WorkflowId), tag.WorkflowRunID(we.RunId)}
+
+		resp, err := a.historyClient.DescribeMutableState(ctx, &historyservice.DescribeMutableStateRequest{
+			NamespaceId: request.NamespaceID,
+			Execution:   &we,
+		})
+
+		switch err.(type) {
+		case nil:
+			if resp.GetCacheMutableState().GetExecutionState().GetState() == enumsspb.WORKFLOW_EXECUTION_STATE_ZOMBIE {
```

Comment:
`if resp.GetCacheMutableState()` this may return nil, use `GetDatabaseMutableState`

Reply:
what are the difference between 2? `GetDatabaseMutableState` also just read from the object?

Reply:
GetCacheMutableState contains the cached version of mutable state, which can be nil (not cached)
GetDatabaseMutableState directly load mutable state from DB